src/client/components/SocialButtons.tsx

```diff
@@ -47,75 +59,106 @@ const iconOverrides = css`
   ></span>
 );
 
-export const SocialButtons = ({
+const SocialButton = ({
+  label,
+  icon,
+  socialProvider,
   queryParams,
-  marginTop,
+  showGap,
+  isNativeApp,
 }: SocialButtonProps) => {
   return (
-    <div css={containerStyles(marginTop)}>
-      <LinkButton
-        priority="tertiary"
-        cssOverrides={[buttonOverrides, iconOverrides]}
-        icon={<SvgGoogleBrand />}
-        href={buildUrlWithQueryParams(
-          '/signin/:social',
-          {
-            social: 'google',
-          },
-          queryParams,
-        )}
-        data-cy="google-sign-in-button"
-        data-link-name="google-social-button"
-      >
-        Google
-      </LinkButton>
-      <Gap />
-      <LinkButton
-        priority="tertiary"
-        cssOverrides={buttonOverrides}
-        icon={<SvgFacebookBrand />}
-        href={buildUrlWithQueryParams(
-          '/signin/:social',
-          {
-            social: 'facebook',
-          },
-          queryParams,
-        )}
-        data-cy="facebook-sign-in-button"
-        data-link-name="facebook-social-button"
-      >
-        Facebook
-      </LinkButton>
-      <Gap />
+    <>
       <LinkButton
         priority="tertiary"
         cssOverrides={[buttonOverrides, iconOverrides]}
-        icon={<SvgAppleBrand />}
+        icon={icon}
         href={buildUrlWithQueryParams(
           '/signin/:social',
           {
-            social: 'apple',
+            social: socialProvider,
           },
           queryParams,
         )}
-        data-cy="apple-sign-in-button"
-        data-link-name="apple-social-button"
+        data-cy={`${socialProvider}-sign-in-button`}
+        data-link-name={`${socialProvider}-social-button`}
       >
-        Apple
+        {socialButtonLabel(label, isNativeApp)}
       </LinkButton>
+      {showGap && <Gap isNativeApp={isNativeApp} />}
+    </>
+  );
+};
+
+const socialButtonLabel = (label: string, isNativeApp?: IsNativeApp) => {
+  const capitalisedLabel = label.charAt(0).toUpperCase() + label.slice(1);
+  if (isNativeApp) {
+    return `Continue with ${capitalisedLabel}`;
+  } else {
+    return capitalisedLabel;
+  }
+};
+
+const socialButtonIcon = (socialProvider: string): React.ReactElement => {
+  switch (socialProvider) {
+    case 'google':
+      return <SvgGoogleBrand />;
+    case 'apple':
+      return <SvgAppleBrand />;
+    case 'facebook':
+      return <SvgFacebookBrand />;
+    default:
+      // null is the officially recommended way to return nothing from a React component,
+      // but LinkButton doesn't accept it, so we return an empty JSX element instead
+      // cf. https://stackoverflow.com/a/47192387
+      return <></>;
+  }
+};
+
+const getButtonOrder = (isNativeApp?: IsNativeApp): string[] => {
+  switch (isNativeApp) {
+    case 'ios':
+      return ['apple', 'facebook', 'google'];
+    case 'android':
+      return ['google', 'facebook', 'apple'];
+    default:
+      return ['google', 'facebook', 'apple'];
```

Comment:
not the point and probably not for this PR, but i don't think facebook should be in the middle, it should be last 😂

Reply:
actually why not this PR?

Reply:
the screenshot you've got in the description clearly shows facebook last

```suggestion
      return ['apple', 'google', 'facebook'];
    case 'android':
      return ['google', 'apple', 'facebook'];
    default:
      return ['google', 'apple', 'facebook'];
```

Reply:
it's a small enough change to make